### PR TITLE
Add a UPI based src template.

### PR DESCRIPTION
**Pull request by markusthoemmes**
The UPI E2E template is focused on running origin e2e tests and nothing else.

This is mostly a copy of the UPI E2E template, mostly copying over the test part of the `src` template and adjusting the top level parameters accordingly.

/assign @wking @jstuever

### CODE REVIEW

File: ci-operator/templates/openshift/installer/cluster-launch-installer-upi-src.yaml [hunk 1]
```diff
@@ -0,0 +1,1303 @@
...
+           "DiskProvisioning": "thin",
+           "MarkAsTemplate": false,
+           "PowerOn": false,
+           "InjectOvfEnv": false,
+           "WaitForIP": false,
+           "Name": "${VM_TEMPLATE}"
+        }
+        EOF
+            cat > /tmp/artifacts/installer/install-config.yaml << EOF
+        apiVersion: v1beta4
```
Comment (jcpowermac) [inline ci-operator/templates/openshift/installer/cluster-launch-installer-upi-src.yaml]: @markusthoemmes apiVersion should be v1.  I need to change that in the other template as well.

**Comment by jstuever** [comment]
Defer to Trevor.
/unassign @jstuever

**Comment by wking** [comment]
> This is mostly a copy of the UPI E2E template, mostly copying over the test part of the src template...

This makes sense to me.  But to try to keep the templates as close as possible, can we make this a series of pivots instead of one commit?  For example, I don't understand what `IMAGE_FORMAT` is about.  Seems to have something to do with logging (e.g. #4788), but it's not obvious to me why you'd need it in `cluster-launch-installer-upi-src.yaml` but not in `cluster-launch-installer-upi-e2e.yaml`.  Same thing for your `setup-google-cloud-sdk` addition; seems like that should go in both UPI templates.  And the Azure / OpenStack `"${CLUSTER_TYPE}"` switches too.

**Comment by markusthoemmes** [comment]
@wking that does make sense to me. I'll see if I can make this easier on the reviewers and I agree that some of these bits look like they should be in the e2e manifest anyway.

**Comment by markusthoemmes** [comment]
Updated the template to pick up my latest commit on master. The diff between e2e and src template is now this:

```diff
12c12
< - name: IMAGE_TESTS
---
> - name: LOCAL_IMAGE_SRC
218c218
<       image: ${IMAGE_TESTS}
---
>       image: ${LOCAL_IMAGE_SRC}
257c257
<         export PATH=/usr/libexec/origin:$PATH
---
>         export PATH=/tmp/shared:$PATH
325,339d324
<         mkdir -p /tmp/output
<         cd /tmp/output
< 
<         function run-upgrade-tests() {
<           openshift-tests run-upgrade "${TEST_SUITE}" --to-image "${RELEASE_IMAGE_LATEST}" \
<             --provider "${TEST_PROVIDER:-}" -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit
<           return 0
<         }
< 
<         function run-tests() {
<           openshift-tests run "${TEST_SUITE}" \
<             --provider "${TEST_PROVIDER:-}" -o /tmp/artifacts/e2e.log --junit-dir /tmp/artifacts/junit
<           return 0
<         }
< 
```

**Pull request merged**